### PR TITLE
update the error codes reported by file_delete

### PR DESCRIPTION
**Pull request by edgargabriel**
This commit fixes the bug reported on the mailing list 

https://github.com/open-mpi/ompi/issues/2232#issuecomment-254370372

@vvenkates27  would you mind reviewing this commit?

### CODE REVIEW

File: ompi/mca/io/ompio/io_ompio_file_open.c [hunk 1]
```diff
@@ -415,9 +415,14 @@ int mca_io_ompio_file_delete (const char *filename,
 
     ret = unlink(filename);
 
-    if (0 > ret && ENOENT != errno ) {
-	opal_output (1, "errno = %d %s\n", errno, strerror(errno));
-        return MPI_ERR_ACCESS;
+    if (0 > ret ) {
+        if ( ENOENT == errno ) {
+//            opal_output (1, "errno = %d %s\n", errno, strerror(errno));
```
Comment (vvenkates27) [inline ompi/mca/io/ompio/io_ompio_file_open.c]: Suggestion: Could remove this comment if it has no significance.

**Comment by jsquyres** [comment]
@edgargabriel Use:

```
git commit --amend -s
```

to get the right signed-off-by line.

**Comment by jsquyres** [comment]
@vvenkates27 Don't forget to use the new Github code review tool next time: https://github.com/blog/2123-more-code-review-tools

**Comment by jsquyres** [comment]
@vvenkates27 @regrant Oh! I see what happened -- @vvenkates27 doesn't have commit perms on the repo, so his review didn't count towards the github-required review.  Gotcha.  I'll provide a review here so that it satisfies the github requirement.

**Pull request merged**